e2etests/test/empty_messages.js

```diff
@@ -0,0 +1,210 @@
+                debug('At service client send callback - error is: ' + sendErr);
+                if (sendErr) {
+                  done(sendErr);
+                } else {
+                  testRendezvous.imDone(serviceClientParticipant);
+                }
+              });
+            }
+          });
+        }
+      });
+    });
+  });
+
+  describe('Over ' + deviceTransport.name + ' using device/eventhub clients - messaging', function () {
+    this.timeout(18000);
+
+    var deviceClient, ehClient, ehReceivers, provisionedDevice;
+
+    before(function (beforeCallback) {
+      createDeviceMethod(function (err, testDeviceInfo) {
+        provisionedDevice = testDeviceInfo;
+        beforeCallback(err);
+      });
+    });
+
+    after(function (afterCallback) {
+      DeviceIdentityHelper.deleteDevice(provisionedDevice.deviceId, afterCallback);
+    });
+
+    beforeEach(function () {
+      ehClient = eventHubClient.fromConnectionString(hubConnectionString);
+      deviceClient = createDeviceClient(deviceTransport, provisionedDevice);
+      ehReceivers = [];
+    });
+
+    afterEach(function (done) {
+      closeDeviceEventHubClients(deviceClient, ehClient, ehReceivers, done);
+    });
+    it('Device sends a message of zero size and it is received by the service', function (done) {
+      var deviceClientParticipant = 'deviceClient';
+      var ehClientParticipant = 'ehClient';
+      var testRendezvous = new Rendezvous(done);
+      this.timeout(120000);
+      //
+      // The size of non payload is calculated based on the value of the messageId value size
+      // and then on the application property value size plus one for the application property name.
+      //
+      var uuidData = uuid.v4();
+      var message = new Message('');
+      message.messageId = uuidData;
+      ehClient.open()
+        .then(function() {testRendezvous.imIn(ehClientParticipant);})
+        .then(ehClient.getPartitionIds.bind(ehClient))
+        .then(function (partitionIds) {
+          return partitionIds.map(function (partitionId) {
+            return ehClient.createReceiver('$Default', partitionId,{ 'startAfterTime' : Date.now()}).then(function(receiver) {
+              ehReceivers.push(receiver);
+              receiver.on('errorReceived', function(err) {
+                done(err);
+              });
+              receiver.on('message', function (eventData) {
+                  if ((eventData.annotations['iothub-connection-device-id'] === provisionedDevice.deviceId) &&
+                      (eventData.systemProperties.messageId === uuidData) &&
+                      (!eventData.body || (eventData.body.length === 0))) {
+                    receiver.removeAllListeners();
+                    testRendezvous.imDone(ehClientParticipant);
+                  } else {
+                    debug('Incoming device id is: '+eventData.systemProperties['iothub-connection-device-id']);
```

Comment:
> '+e [](start = 49, length = 4)

nit, spacing

Reply:
Fixed.

---
In reply to: [227565943](https://github.com/Azure/azure-iot-sdk-node/pull/401#discussion_r227565943) [](ancestors = 227565943)